cli/command/image/trust.go

```diff
@@ -301,8 +299,11 @@ func trustedPull(ctx context.Context, cli *command.DockerCli, repoInfo *registry
 		}
 
 		// If reference is not trusted, tag by trusted reference
-		if !r.reference.HasDigest() {
-			tagged, err := reference.WithTag(repoInfo, r.reference.String())
+		// TODO(aaronl): Is this code really necessary? Will notary ever
```

Comment:
Are you asking whether the check is necessary since Notary is always mapping untrusted tags to digest?  This block should always end up getting called. In that case where the r.name originally was a digest, it shouldn't even hit this section since the Notary lookup could be completely bypassed.

> Will notary ever give us a target that maps a digest to a digest?

That was not the intent of this block but I don't think it was something that was ruled out (mapping different hash algorithms to a single canonical digest).